scarches/dataset/scpoli/anndata.py

```diff
@@ -76,27 +76,31 @@ def __init__(self,
             self.cell_types = np.stack(self.cell_types).T
             self.cell_types = torch.tensor(self.cell_types, dtype=torch.long)
 
-    def __getitem__(self, index):
+    def __getitems__(self, indices):
```

Comment:
The core implementation is not duplicated. `__getitem__` now calls `__getitems__` which accepts both list indices and single indices.
The reason that we have to maintain both interfaces is that pytorch data loader originally uses `__getitem__`. But if you also provide `__getitems__`. then it will use it, and we benefit from the parallel fetching of samples. I am not sure about the newer versions of pytorch but the one scArches is using does not automatically fetch multiple samples if  `__getitems__` interface is not available.
You can also check [this issue](https://github.com/pytorch/pytorch/issues/107218) in pytorch repo.

Reply:
I see, thanks! Didn't know it about pytorch